modules/tensor_mechanics/src/materials/ComputeFiniteStrain.C

```diff
@@ -174,3 +128,91 @@ ComputeFiniteStrain::computeQpStrain()
   _mechanical_strain[_qp] = _rotation_increment[_qp] * _mechanical_strain[_qp] * _rotation_increment[_qp].transpose();
   _total_strain[_qp] = _rotation_increment[_qp] * _total_strain[_qp] * _rotation_increment[_qp].transpose();
 }
+
+void
+ComputeFiniteStrain::computeQpIncrements(RankTwoTensor & total_strain_increment, RankTwoTensor & rotation_increment)
+{
+  if (_decomposition_method == "Rashid")
```

Comment:
Switches "might" be faster under the right conditions but this is a big, ugly, nasty slow string comparison so there are no speed differences here.

What we should be doing for cases like this is creating a scoped enum (POD type) that corresponds to the options in the MooseEnum. We can directly set the enum from the MooseEnum when it's read in. Then inside of these "hot" qp loops you are doing a comparison like this which are way faster than string comparisons.

Your code looks almost the same:
```
  if (_decomposition_method == Method::Rashid)
```

If you have questions about how to do this, please let me know.